packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -119,10 +127,66 @@ export class SerializedStateManager {
 		} else {
 			const { baseSnapshot, snapshotBlobs } = this.pendingLocalState;
 			this.snapshot = { baseSnapshot, snapshotBlobs };
+			this.refreshSnapshot ??= (async () => {
+				this.latestSnapshot = await getLatestSnapshotInfo(
+					this.mc,
+					this.storageAdapter,
+					supportGetSnapshotApi,
+				);
+				this.refreshPendingAttributes();
+			})();
+
 			return { baseSnapshot, version: undefined };
 		}
 	}
 
+	/**
+	 * Updates class snapshot and processedOps with the latest fetched snapshot
+	 *
+	 * @param baseSnapshot -
+	 */
+	private refreshPendingAttributes() {
+		if (this.latestSnapshot === undefined) {
+			return;
+		}
+		const snapshotSequenceNumber = this.latestSnapshot?.snapshotSequenceNumber;
+		if (this.processedOps.length === 0) {
+			// weird edge case in which we don't have to do nothing to processedOps
```

Comment:
this comment isn't quite right. i believe its not that uncommon of a case, basically client goes offline after getting snapshot, but before it receives any remote ops. would be good to test this case.